app/services/preserved_object_handler.rb

```diff
@@ -171,16 +171,22 @@ def update_version(checksums_validated = false)
       update_online_version(new_status, true, checksums_validated)
     end
 
-    results.report_results
+    results.report_results(@logger, comp_moab)
   end
 
   def pres_object
     @pres_object ||= PreservedObject.find_by!(druid: druid)
+  rescue ActiveRecord::RecordNotFound => e
+    results.add_result(AuditResults::DB_OBJ_DOES_NOT_EXIST, e.inspect)
+    nil
```

Comment:
these rescue blocks became necessary as we need to do more with these db objects now as part of `report_results`